web-ui/view/documentation.ml

```diff
@@ -103,3 +106,177 @@ let getting_started =
+                    "OCaml-CI is a continuous integration (CI) tool for OCaml \
+                     projects. This development practice encourages developers \
+                     to regularly integrate code into a shared repository. \
+                     Commits are verified by an automated build, allowing \
+                     teams to detect and fix problems early.";
+                ];
+              p
+                [
+                  txt "OCaml-CI uses metadata from the project’s opam and ";
+                  code [ txt "dune" ];
+                  txt
+                    " files to work out what to build, and it also uses \
+                     caching to make builds fast. It takes the information in \
+                     the project's opam files to automatically test against \
+                     multiple OCaml versions and OS platforms.";
+                ];
+              p
+                [
+                  txt
+                    "Simply put, OCaml-CI just tells you if a project is \
+                     compatible with OCaml 5 or not.";
+                ];
+              p
+                [
+                  txt "In ";
+                  a
+                    ~a:[ a_href "https://ci.ocamllabs.io/getting-started" ]
+                    [ txt "Getting Started" ];
+                  txt
+                    ", you installed the the OCaml-CI app, which follows this \
+                     pattern when deployed:";
+                ];
+              ul
+                [
+                  li
+                    [
+                      txt
+                        "First, it retreives the list of organisations that \
+                         have installed it.";
+                    ];
+                  li
+                    [
+                      txt
+                        "For each organisation, it gets the list of \
+                         repositories to check.";
+                    ];
+                  li
+                    [
+                      txt
+                        "For each repository, it gets the branches and PRs to \
+                         check.";
+                    ];
+                  li
+                    [
+                      txt
+                        "For each target, it fetches the head commit, \
+                         generates a Dockerfile, and builds it.";
+                    ];
+                ];
+              p
+                [
+                  txt "The generated Dockerfile first adds all the ";
+                  code [ txt "*.opam" ];
+                  txt
+                    " files found in the project, then uses opam to install \
+                     all the dependencies, and finally adds the rest of the \
+                     source files. This means that rebuilds are often very \
+                     fast, because Docker will reuse the previously cached \
+                     build step as long as the opam files don’t change.";
```

Comment:
Can't we add a sentence saying that it tries to install all the opam files at the same time and pin the version to the local version? This way, we ensure to discard all exotic configurations that require extra steps.

Reply:
This seems important but I don't understand this enough to write this simply and correctly. Can you please suggest an edit?

Reply:
An example would be clearer here.

Given a project with the following opam files
```
bondi.opam
bondi-lwt.opam
bondi-async.opam
```
OCaml-CI will run an opam solve for all of the dependencies in all the opam files together (implied by the opam install but perhaps not clear enough). So if you depend on OCaml 5 in `bondi-lwt.opam` then a build will only happen on that version of OCaml, even if the other two packages could run on earlier versions of OCaml.

See https://github.com/ocurrent/ocaml-ci/issues/470 for the issue suggesting an alternate build strategy.

Reply:
> OCaml-CI will run an opam solve for all of the dependencies in all the opam files together
Not with the right wording but that's what I meant ^^

Reply:
How about: 

OCaml-CI will run an opam solve for all dependencies in every opam file simultaneously.